packages/tss/src/tss.ts

```diff
@@ -0,0 +1,790 @@
+    const newTSSCommits = [
+      Point.fromJSON(tssPubKey),
+      Point.fromJSON({ x: secondCommit.getX().toString(16, 64), y: secondCommit.getY().toString(16, 64) }),
+    ];
+    const factorEncs: {
+      [factorPubID: string]: FactorEnc;
+    } = {};
+    for (let i = 0; i < refreshResponses.length; i++) {
+      const refreshResponse = refreshResponses[i];
+      factorEncs[refreshResponse.factorPub.x.padStart(64, "0")] = {
+        type: "hierarchical",
+        tssIndex: refreshResponse.targetIndex,
+        userEnc: refreshResponse.userFactorEnc,
+        serverEncs: refreshResponse.serverFactorEncs,
+      };
+    }
+
+    this.metadata.addTSSData({
+      tssKeyType: this._tssKeyType,
+      tssTag: this.tssTag,
+      tssNonce: tssNonce + 1,
+      tssPolyCommits: newTSSCommits,
+      factorPubs,
+      factorEncs,
+    });
+    if (updateMetadata) await this._syncShareMetadata();
+  }
+
+  /**
+   * Derives the account nonce for the specified account index.
+   */
+  computeAccountNonce(index?: number): BN {
+    if (!index || index === 0) {
+      return new BN(0);
+    }
+
+    if (this._tssKeyType === KeyType.ed25519) {
+      throw new Error("account index not supported with ed25519");
+    }
+
+    // generation should occur during tkey.init, fails if accountSalt is absent
+    if (!this._accountSalt) {
+      throw Error("account salt undefined");
+    }
+    let accountHash = keccak256(Buffer.from(`${index}${this._accountSalt}`));
+    if (accountHash.length === 66) accountHash = accountHash.slice(2);
+    return new BN(accountHash, "hex").umod(this._tssCurve.n);
+  }
+
+  /**
+   * Reconstructs the TKey and finalize intialization.
+   */
+  async reconstructKey(_reconstructKeyMiddleware?: boolean): Promise<ReconstructedKeyResult> {
+    const k = await super.reconstructKey(_reconstructKeyMiddleware);
+
+    // only valid for use Tss
+    // assign account salt from tKey store if it exists
+    if (Object.keys(this.metadata.tssPolyCommits).length > 0) {
+      const accountSalt = (await this.getTKeyStoreItem(TSS_MODULE, "accountSalt")) as IAccountSaltStore;
+      if (accountSalt && accountSalt.value) {
+        this._accountSalt = accountSalt.value;
+      } else {
+        const newSalt = generateSalt(this._tssCurve);
+        await this._setTKeyStoreItem(TSS_MODULE, {
+          id: "accountSalt",
+          value: newSalt,
+        } as IAccountSaltStore);
+        this._accountSalt = newSalt;
+        // this is very specific case where exisiting user do not have salt.
+        // sync metadata to cloud to ensure salt is stored incase of manual sync mode
+        // new user or importKey should not hit this cases
+        // NOTE this is not mistake, we force sync for this case
+        if (this.manualSync) await this.syncLocalMetadataTransitions();
+      }
+    }
+
+    return k;
+  }
+
+  /**
+   * Adds a factor key to the set of authorized keys.
+   */
+  public async addFactorPub(tssOptions: {
```

Comment:
I think we should have copyFactorPub function

Reply:
so you mean instead of having [`copyOrCreateShare`](https://github.com/Web3Auth/mpc-core-kit/blob/36e608b73ed12f90160bd8ac2e4c26343f3d1991/src/mpcCoreKit.ts#L1080) in core kit, we move it here? or just create a `copyFactorFunction`?

Reply:
added an option for toggling whether shares should be refreshed or not.